From 1b918d04f541d201a2fa43d371ecbb1318719d4c Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <Michal@Hrusecky.net>
Date: Tue, 23 Jan 2018 23:55:00 +0100
Subject: [PATCH] ubus: Hackilly add Stepans python bindings

There are currently two problems with this solution:

1) I don't have a working python so I have to hack my way around it

2) All the sources are currently part of patch

  * Best solution would be push it upstream
  * At worst we can use src directory

Signed-off-by: Michal Hrusecky <Michal@Hrusecky.net>
---
 package/system/ubus/Makefile | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/package/system/ubus/Makefile b/package/system/ubus/Makefile
index 523c362..d39b918 100644
--- a/package/system/ubus/Makefile
+++ b/package/system/ubus/Makefile
@@ -47,11 +47,34 @@ define Package/libubus-lua
   TITLE:=Lua binding for the OpenWrt RPC client
 endef
 
+define Package/python-ubus
+  SUBMENU:=Python
+  SECTION:=lang
+  CATEGORY:=Languages
+  TITLE:=python-ubus
+  DEPENDS:=+libubus +libblobmsg-json +python-light
+endef
+
 TARGET_CFLAGS += -I$(STAGING_DIR)/usr/include -flto
 TARGET_LDFLAGS += -flto
 
+define Build/Compile/fake_python_sh
+	echo -e '#!/bin/sh\nLD_LIBRARY_PATH="$(STAGING_DIR_HOSTPKG)/lib" $(STAGING_DIR_HOSTPKG)/bin/python "$$$$@"' > '$(STAGING_DIR_HOSTPKG)/bin/python.sh'
+	chmod a+rx '$(STAGING_DIR_HOSTPKG)/bin/python.sh'
+endef
+
+Hooks/Configure/Pre += Build/Compile/fake_python_sh
+
+ifdef CONFIG_PACKAGE_python-ubus
+CMAKE_OPTIONS = \
+	-DLUAPATH=/usr/lib/lua  \
+	-DBUILD_PYTHON=ON \
+	-DPYTHON='$(STAGING_DIR_HOSTPKG)/bin/python.sh' \
+	-DPYTHON_PACKAGE_VERSION=$(PKG_VERSION)
+else
 CMAKE_OPTIONS = \
 	-DLUAPATH=/usr/lib/lua
+endif
 
 define Package/ubus/install
 	$(INSTALL_DIR) $(1)/bin
@@ -73,7 +96,13 @@ define Package/libubus-lua/install
 	$(CP) $(PKG_BUILD_DIR)/lua/ubus.so $(1)/usr/lib/lua/
 endef
 
+define Package/python-ubus/install
+	$(INSTALL_DIR) $(1)/usr/lib/python2.7/site-packages
+	$(CP) $(PKG_BUILD_DIR)/python/build/*/ubus.so $(1)/usr/lib/python2.7/site-packages
+endef
+
 $(eval $(call BuildPackage,libubus))
 $(eval $(call BuildPackage,libubus-lua))
 $(eval $(call BuildPackage,ubus))
 $(eval $(call BuildPackage,ubusd))
+$(eval $(call BuildPackage,python-ubus))
-- 
2.18.0

